darwin/dataset/upload_manager.py

```diff
@@ -338,6 +338,14 @@ def callback(file_name, file_total_bytes, file_bytes_sent):
                     file_complete.add(file_name)
                     progress_callback(self.pending_count, 1)
 
+        if max_workers:
+            if max_workers < 1:
+                raise ValueError("max_workers must be greater than 0")
+            elif max_workers > concurrent.futures.ThreadPoolExecutor()._max_workers:
+                raise ValueError(
```

Comment:
I'd suggest that here, instead of raising a `ValueError` exception, we can just set the `max_workers` to be the max workers possible on the machine.

Reply:
I would actually rather to raise here, rather then implicitly "fix" things behind the scene, so IMO we can keep the code we have here.

Reply:
Alright cool!